### PR TITLE
feat: Support __Host- prefixed session cookies for sticky sessions

### PR DESCRIPTION
**Pull request by hoffmaen**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
 - Gorouter now recognises `__Host-`-prefixed variants of configured sticky session cookie names (e.g. `__Host-JSESSIONID`), both in application responses and client requests
  - Uses exact prefix matching (`__Host-`) per [RFC 6265bis](https://www.rfc-editor.org/rfc/draft-ietf-httpbis-rfc6265bis-19.html#name-the-__host-prefix), which mandates this canonical casing for user agents — no case-insensitive matching needed
  - No additional configuration required; the `__Host-` prefix is handled automatically for every name in `router.sticky_session_cookie_names`

Motivation
---------------
The `__Host-` cookie prefix ensures that session cookies are bound to the exact origin host and are never sent to subdomains, preventing session cookie leakage to other applications on shared domains. This prefix also enforces that cookies are only set over HTTPS, without a `Domain` attribute, and with `Path=/` — supporting it in gorouter's sticky session handling allows applications to leverage these stricter security guarantees while retaining session affinity.

Backward Compatibility
---------------
Breaking Change? **No**

Note on AI usage
---------------
Parts of this code and tests were developed with assistance from Claude Code (claude-opus-4-20250514).

### CODE REVIEW

File: docs/03-how-to-use-session-affinity.md [hunk 1]
```diff
@@ -106,6 +106,37 @@ for the old non-partitioned `__VCAP_ID__` cookie alongside the new partitioned o
 
 <img src="images/sticky_sessions_chips_migration.png" alt="Sticky Sessions - CHIPS migration sequence" width="800">
 
+### Does Gorouter support `__Host-` prefixed session cookies?
+Yes. [RFC 6265bis](https://www.rfc-editor.org/rfc/draft-ietf-httpbis-rfc6265bis-19.html#name-the-__host-prefix) defines
```
Comment (a18e) [inline docs/03-how-to-use-session-affinity.md]: This link seems broken search doesn't return anything, how about https://datatracker.ietf.org/doc/draft-ietf-httpbis-rfc6265bis/ ?
(though that one will probably also change once it's no longer draft)

File: docs/03-how-to-use-session-affinity.md [hunk 2]
```diff
@@ -106,6 +106,37 @@ for the old non-partitioned `__VCAP_ID__` cookie alongside the new partitioned o
...
+
+In practice this is not a concern: unlike CHIPS migration, there is no need to set both cookies in
+the same response. Because `JSESSIONID` and `__Host-JSESSIONID` are distinct cookie names in the
+browser's jar, the expected migration path is for the application to simply stop setting
+`JSESSIONID` and start setting `__Host-JSESSIONID` — the old cookie expires naturally.
+
+Note: if an application were to set a new `__Host-JSESSIONID` alongside a delete (`Max-Age=0`) for
+the old `JSESSIONID` in the same response, both would produce a `__VCAP_ID__` in the same cookie
+jar partition. Depending on processing order, the browser could apply the delete `__VCAP_ID__`
+after the new one, effectively removing it.
```
Comment (a18e) [inline docs/03-how-to-use-session-affinity.md]: Maybe add what this actually means for a developer, e.g. "This could mean breaking session stickyness for the time of migration from JSESSIONID to _Host-...?"

File: src/code.cloudfoundry.org/gorouter/handlers/helpers.go [hunk 3]
```diff
@@ -87,9 +87,15 @@ func GetStickySession(request *http.Request, stickySessionCookieNames config.Str
...
-	for stickyCookieName := range stickySessionCookieNames {
-		if _, err := request.Cookie(stickyCookieName); err == nil {
+
+	// Try choosing a backend using sticky session.
+	// Also match the "__Host-" prefixed variant of each configured cookie name (RFC 6265bis).
+	for _, c := range request.Cookies() {
+		name := c.Name
+		if strings.HasPrefix(name, "__Host-") {
+			name = name[7:]
+		}
```
Comment (a18e) [inline src/code.cloudfoundry.org/gorouter/handlers/helpers.go]:
```suggestion
		name := strings.TrimPrefix(c.Name, "__Host-")
```
Wouldn't this be simpler? see also below

File: src/code.cloudfoundry.org/gorouter/proxy/round_tripper/proxy_round_tripper.go [hunk 4]
```diff
@@ -597,13 +597,26 @@ func getSessionCookies(response *http.Response, stickySessionCookieNames config.
...
+	if _, ok := stickySessionCookieNames[cookieName]; ok {
+		return true
+	}
+	if strings.HasPrefix(cookieName, "__Host-") {
+		_, ok := stickySessionCookieNames[cookieName[7:]]
+		return ok
+	}
+	return false
+}
+
```
Comment (a18e) [inline src/code.cloudfoundry.org/gorouter/proxy/round_tripper/proxy_round_tripper.go:614]:
```suggestion
		// also match __Host- prefix
		name := strings.TrimPrefix(cookie.Name, "__Host-")
		if _, ok := stickySessionCookieNames[name]; ok {

			sessionCookies = append(sessionCookies, cookie)
		}
	}
	return sessionCookies, nil
}
```

Comment (hoffmaen) [reply src/code.cloudfoundry.org/gorouter/proxy/round_tripper/proxy_round_tripper.go:614]: I simplified it to use `strings.TrimPrefix` but decided to keep `IsSessionCookie` as a separate exported function — it has dedicated unit tests that cover edge cases (wrong casing, missing hyphen, unknown names), which I find easier to maintain as a standalone helper.

**Review by a18e — APPROVED** [review]
LGTM
